src/GitHub.App/Services/PullRequestService.cs

```diff
@@ -95,5 +255,33 @@ public IObservable<string> GetPullRequestTemplate(ILocalRepositoryModel reposito
             return ret;
         }
 
+        void EnsurePullRefSpecExists(IRepository repo)
+        {
+            var spec = "+refs/pull/*/head:refs/remotes/origin/pr/*";
+            var origin = repo.Network.Remotes["origin"];
+            var existing = origin.FetchRefSpecs.FirstOrDefault(x => x.Specification == spec);
+
+            if (existing == null)
+            {
+                repo.Network.Remotes.Update(origin, x => x.FetchRefSpecs.Add(spec));
+            }
+        }
+
+        static string GetSafeBranchName(string name)
```

Comment:
Thinking about this a little over the weekend, I think the current implementation could cause problems with non-Latin character sets. So I guess that if the title has no Latin characters we should fall back to simply `pr/123`.